lib/modules/apostrophe-events-calendar-widgets/lib/api.js

```diff
@@ -0,0 +1,44 @@
+var moment = require('moment');
+var _ = require('lodash');
+
+module.exports = function(self, options) {
+  self.getEvents = async function(req) {
+    var month;
+    if (req.query && !req.body.month) {
+      month = moment().format('YYYY-MM');
+    } else {
+      month = req.body.month
+    }
+
+    var max = moment(month).add({ days: 15, months: 1 }).format('YYYY-MM-DD');
+    var min = moment(month).subtract(15, 'd').format('YYYY-MM-DD');
+    var query = { startDate: { $lte: max, $gte: min } }
+    var projection = _.merge((self.options.projection || {}), { startDate:1, title: 1, url: 1, type:1, slug:1 });
+    var events = await self.apos.modules[self.piecesModuleName].find(req, query, projection).toArray();
+
+    events.forEach(event => {
+      event.date = event.startDate      
+    });
+
+    var clndrOptions = {
+      template: self.render(req, '_clndr', {}),
+      daysOfTheWeek: ['Su', 'M', 'T', 'W', 'Th', 'F', 'S'],
+      events: events,
+      moment: moment,
```

Comment:
This is a server side library of code, I think it won't do anything useful to include it in a JSON response unless I've misunderstood what it is.

Reply:
yeah i had copy pasta'd this from another implementation .. seems to work without it (when moment is part of a non-lean apos site). Should i push it as its own asset alongside clndr? Controlled by the same `pushClndr` flag to opt out?

Reply:
broke this out as its own option to push moment.js.

if apostrophe-assets is set to `lean: true` and your calendar module is NOT set to `pushMoment: false`, you get moment along side the other assets.